src/logging.cpp

```diff
@@ -196,29 +199,31 @@ std::string BCLog::Logger::LogTimestampStr(const std::string &str)
     } else
         strStamped = str;
 
-    if (!str.empty() && str[str.size()-1] == '\n')
-        m_started_new_line = true;
-    else
-        m_started_new_line = false;
-
     return strStamped;
 }
 
 void BCLog::Logger::LogPrintStr(const std::string &str)
 {
-    std::string strTimestamped = LogTimestampStr(str);
+    std::string strPrefixed = str;
+
+    if (m_log_threadnames && m_started_new_line)
+        strPrefixed.insert(0, "[" + thread_util::GetInternalName() + "] ");
+
+    strPrefixed = LogTimestampStr(strPrefixed);
+
+    m_started_new_line = (!str.empty() && str[str.size()-1] == '\n');
```

Comment:
In commit "threads: prefix log messages with thread names" (4fba3ead9811ebe47161e9e7f535a7cc88403cf5)

Some unneeded parentheses here